torch/csrc/jit/mobile/promoted_prim_ops.cpp

```diff
@@ -92,6 +92,15 @@ void device(Stack& stack) {
   push(stack, pop(stack).toTensor().device());
 }
 
+void device_with_index(Stack& stack) {
+  at::Tensor device;
+  int index;
+  pop(stack, device);
+  pop(stack, index);
+  device.device().set_index(index);
+  push(stack, device);
```

Comment:
```suggestion
  at::Tensor tensor;
  int index;
  pop(stack, tensor);
  pop(stack, index);
  auto device = tensor.device();
  device.set_index(index);
  push(stack, device);
```

Reply:
Thanks, will have to refactor so that inputs are `string, int` instead of `Tensor, int`